src/Core/TileMesh.js

```diff
@@ -81,9 +86,30 @@ TileMesh.prototype.changeState = function changeState(state) {
         delete this.material.defines.DEPTH_MODE;
     }
 
+    this._state = state;
+
     this.material.needsUpdate = true;
 };
 
+function applyChangeState(n, s) {
+    if (n.changeState) {
+        n.changeState(s);
+    }
+}
+
+TileMesh.prototype.pushRenderState = function pushRenderState(state) {
+    if (this._state == state) {
+        return () => { };
+    }
+
+    const oldState = this._state;
+    this.traverse(n => applyChangeState(n, state));
```

Comment:
i don't know whether to check if `n` is an instance of `TileMesh`?

Reply:
It's not needed since `applyChangeState` checks for `changeState` existence.

Reply:
yes